test/e2e/run_privileged_test.go

```diff
@@ -131,6 +131,30 @@ var _ = Describe("Podman privileged container tests", func() {
 		Expect(len(session.OutputToStringArray())).To(BeNumerically(">", 20))
 	})
 
+	It("podman privileged should restart after host devices change", func() {
+		containerName := "privileged-restart-test"
+		SkipIfRootless("Cannot create devices in /dev in rootless mode")
+		Expect(os.MkdirAll("/dev/foodevdir", os.ModePerm)).To(BeNil())
+
+		mknod := SystemExec("mknod", []string{"/dev/foodevdir/null", "c", "1", "3"})
+		mknod.WaitWithDefaultTimeout()
+		Expect(mknod).Should(Exit(0))
+
+		session := podmanTest.Podman([]string{"run", "--name=" + containerName, "--privileged", "-it", fedoraMinimal, "ls", "/dev"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		deviceFiles := session.OutputToStringArray()
+
+		os.RemoveAll("/dev/footdevdir")
+		session = podmanTest.Podman([]string{"start", "--attach", containerName})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		deviceFilesAfterRemoval := session.OutputToStringArray()
+		Expect(len(deviceFiles)).To(Not(Equal(len(deviceFilesAfterRemoval))))
```

Comment:
is there way way we can check the actual output rather than the length?

Reply:
I can do `Expect(deviceFiles).To(Not(Equal(deviceFilesAfterRemoval)))`

Reply:
That ought to be safe. What we're testing for is that any change in device nodes does not prevent container restart; if the two don't match exactly, a change in device nodes has occurred, and if the container restarted successfully the PR worked.

Reply:
yeah do that or `Expect(deviceFiles).To(Not(ContainSubstring(....)))`